ext/nio4r/monitor.c

```diff
@@ -117,6 +125,52 @@ static VALUE NIO_Monitor_initialize(VALUE self, VALUE io, VALUE interests, VALUE
     return Qnil;
 }
 
+//Still under construction
+static VALUE NIO_Monitor_register(VALUE self, VALUE io, VALUE interests){
+    return Qnil;
+}
+
+static VALUE NIO_Monitor_setInterests(VALUE self, VALUE interests){
+    //Check Whether the Monitor is already Closed?
+    if(NIO_Monitor_is_closed(self) == Qfalse){
+       struct NIO_Monitor *monitor;
+        ID interests_id;
+
+        //Since these are not declared in the C version with CRuby 2.0.0
+        int stdout = 1;
+        int stdin = 0;
+
+        interests_id = SYM2ID(interests);
+        Data_Get_Struct(self, struct NIO_Monitor, monitor);
+        int mask = 0;
+        if(interests_id == rb_intern("r")) {
+            monitor->interests = EV_READ;
+            mask = stdin;
+        } else if(interests_id == rb_intern("w")) {
+            monitor->interests = EV_WRITE;
+            mask = stdout;
+        } else if(interests_id == rb_intern("rw")) {
+            monitor->interests = stdin | stdout;
+            mask = stdin | stdout;
+        } else {
+            rb_raise(rb_eArgError, "invalid event type %s (must be :r, :w, or :rw)",
+                RSTRING_PTR(rb_funcall(interests, rb_intern("inspect"), 0, 0)));
+        }
+        rb_ivar_set(self, rb_intern("interests"), interests);
+        //Changing the values in monitor struct
+        ev_io_stop(monitor->selector->ev_loop, &monitor->ev_io);
+        //Acknowledging the libev about the change
+        ev_io_set(&monitor->ev_io, mask, monitor->interests);
```

Comment:
Hmm, I'm surprised this works. I would think you would need to pass `monitor->ev_io.fd` in instead of `mask`.

I'm not sure why you're trying to pass `stdin` / `stdout`?